NotificationSystem/NotificationSystem.Tests.Unit/PostToOrcasiteTests.cs

```diff
@@ -11,20 +11,29 @@ public class PostToOrcasiteTests
     {
         /// <summary>
         /// This test mocks the Orcasite API and verifies that a detection can be posted successfully.
+        /// It also verifies that the mock HTTP client is actually called when processing documents.
         /// </summary>
-        /// <returns></returns>
         [Fact]
         public async Task PostToOrcasite_ProcessDocumentsAsync()
         {
             var mockOrcasiteLogger = new Mock<ILogger<OrcasiteHelper>>();
-            OrcasiteHelper orcasiteHelper = OrcasiteTestHelper.GetMockOrcasiteHelper(mockOrcasiteLogger.Object);
+            var (orcasiteHelper, mockHttp, getFeedsRequest, postDetectionRequest) = OrcasiteTestHelper.GetMockOrcasiteHelperWithRequestVerification(mockOrcasiteLogger.Object);
             List<JsonElement> documents = OrcasiteTestHelper.GetSampleOrcaHelloDetections();
 
             // Process it like the Azure function would.
             var mockFunctionLogger = new Mock<ILogger<PostToOrcasite>>();
             var postToOrcasite = new PostToOrcasite(orcasiteHelper, mockFunctionLogger.Object);
             bool ok = await postToOrcasite.ProcessDocumentsAsync(documents);
+            
+            // Assert that the operation succeeded.
             Assert.True(ok, "PostToOrcasite.ProcessDocumentsAsync failed");
+            
+            // Verify that all expected HTTP calls were made.
+            mockHttp.VerifyNoOutstandingExpectation();
```

Comment:
How does this `VerifyNoOutstandingExpectation` work? Where are the expectations set?